atuin-common/src/record.rs

```diff
@@ -162,21 +178,30 @@ impl RecordIndex {
 
         // account for that!
         for (host, tag_map) in other.hosts.iter() {
-            for (tag, tail) in tag_map.iter() {
+            for (tag, idx) in tag_map.iter() {
                 match self.get(*host, tag.clone()) {
                     // If we have this host/tag combo, the comparison and diff will have already happened above
                     Some(_) => continue,
 
                     None => ret.push(Diff {
                         host: *host,
                         tag: tag.clone(),
-                        tail: *tail,
+                        remote: Some(*idx),
+                        local: None,
                     }),
                 };
             }
         }
 
-        ret.sort_by(|a, b| (a.host, a.tag.clone(), a.tail).cmp(&(b.host, b.tag.clone(), b.tail)));
+        // Stability is a nice property to have
+        ret.sort_by(|a, b| {
+            (a.host, a.tag.clone(), a.local, a.remote).cmp(&(
+                b.host,
+                b.tag.clone(),
+                b.local,
+                b.remote,
+            ))
```

Comment:
```suggestion
            std::cmp::Ord::cmp(
                &(a.host, &a.tag, a.local, a.remote),
                &(b.host, &b.tag, b.local, b.remote),
            )
```

alternatively, `#[derive(PartialEq, PartialOrd, Eq, Ord)]` on Diff and just `ret.sort()`.